src/wallet/types.ts

```diff
@@ -13,11 +14,23 @@ export type RootKeyPair = {
 
 type FclAccountInfo = Awaited<ReturnType<typeof fcl.account>>
 
-export type AccountInfoOnChain = Omit<FclAccountInfo, 'keys'> & {
+type AccountInfoOnChain = Omit<FclAccountInfo, 'keys'> & {
   pubKeyInfo: FclAccountInfo['keys'][number]
 }
 
 export type AccountInfo = AccountInfoOnChain & {
   privKey: PrivKey
   web3authUserInfo: Web3AuthUserMetadata
 }
+
+export type WalletActionsCallbacks = {
+  onCreateAccount: {
```

Comment:
I'd consider merging `onCreateAccount` and `onLoginFail` into a single `onLoginStatusChange` callback which would expect an object like `{status: 'start' | 'success' | 'error' | 'logged_out'} | {status: 'error', error: unknown}` which would IMHO be more stratightforward to use/understand as the whole login lifecycle would be captured in a single entry point.